packages/plugins/prometheus/README.md

```diff
@@ -532,6 +538,73 @@ const getEnveloped = envelop({
 })
 ```
 
+### Configure metric phases
+
+Each metric observes timing or events in different phases of the GraphQL request pipeline.
+
+You can configure which phases are observed for a given metric by providing an array of phases
+instead of `true` for any metric configuration. You can also configure the phases when using custom
+metrics factories by providing the `phases` option.
+
+By default, all available phases are enabled when the metric is enabled.
+
+```ts
+import { execute, parse, specifiedRules, subscribe, validate } from 'graphql'
+import { envelop, useEngine } from '@envelop/core'
+import { usePrometheus } from '@envelop/prometheus'
+
+const getEnveloped = envelop({
+  plugins: [
+    useEngine({ parse, validate, specifiedRules, execute, subscribe }),
+    usePrometheus({
+      metrics: {
+        graphql_envelop_phase_error: ['execute', 'subscribe'] // only trace errors of execute and subscribe phases
+      }
+    })
+  ]
+})
+```
+
+### Skip observation based on request context
+
+To save bandwidth or storage, you can reduce the amount of reported values by filtering which events
+are observed based on the request context.
+
+For example, you can only monitor a subset of operations, because they are critical or that you want
+to debug it's performance:
+
+```ts
+import { execute, parse, specifiedRules, subscribe, validate } from 'graphql'
+import { envelop, useEngine } from '@envelop/core'
+import { usePrometheus } from '@envelop/prometheus'
+
+const TRACKED_OPERATION_NAMES = [
+  // make a list of operation that you want to monitor
+]
+
+const getEnveloped = envelop({
+  plugins: [
+    useEngine({ parse, validate, specifiedRules, execute, subscribe }),
+    usePrometheus({
+      metrics: {
+        graphql_yoga_http_duration: createHistogram({
+          registry,
+          histogram: {
+            name: 'graphql_yoga_http_duration',
+            help: 'Time spent on HTTP connection',
+            labelNames: ['operation_name']
+          },
+          fillLabelsFn: ({ operationName }, _rawContext) => ({
+            operation_name: operationName
+          }),
+          shouldObserve: context => TRACKED_OPERATIONS.includes(context?.params?.operationName)
```

Comment:
I like it! 

But I think the example should be updated:

```suggestion
          shouldObserve: (_, context) => TRACKED_OPERATIONS.includes(context?.params?.operationName)
```

because I think the signature is the same as `fillLabelsFn`.